configure.ac

```diff
@@ -686,20 +686,10 @@ if test x"${PKG_CONFIG}" = x""; then
     AC_MSG_ERROR(You need pkgconfig installed in order to build ${PACKAGE})
 fi
 
-if
-    $PKG_CONFIG --exists glib-2.0
-then
-    GLIBCONFIG="$PKG_CONFIG glib-2.0"
-else
-    set -x
-    echo PKG_CONFIG_PATH=$PKG_CONFIG_PATH
-    $PKG_CONFIG --exists glib-2.0; echo $?
-    $PKG_CONFIG --cflags glib-2.0; echo $?
-    $PKG_CONFIG glib-2.0; echo $?
-    set +x
-    AC_MSG_ERROR(You need glib2-devel installed in order to build ${PACKAGE})
-fi
-AC_MSG_RESULT(using $GLIBCONFIG)
+# Require glib 2.16.0 (2008-03) or later for g_hash_table_iter_init() etc.
+PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.16.0],
+                  [CPPFLAGS="${CPPFLAGS} ${GLIB_CFLAGS}"
+                   LIBS="${LIBS} ${GLIB_LIBS}"])
```

Comment:
Just noticed this, thanks for demessifying.

For future iterations, the ideal remains to be avoidance
of spoiling of generic flags and using granular prefixed
ones instead as appropriate.